templates/components/settings/python/settings.py

```
@@ -55,25 +60,29 @@ def init_openai():
 def init_azure_openai():
     from llama_index.core.constants import DEFAULT_TEMPERATURE
-    from llama_index.embeddings.azure_openai import AzureOpenAIEmbedding
-    from llama_index.llms.azure_openai import AzureOpenAI
+
+    try:
+        from llama_index.embeddings.azure_openai import AzureOpenAIEmbedding
+        from llama_index.llms.azure_openai import AzureOpenAI
+    except ImportError:
+        raise ImportError(
+            "Azure OpenAI support is not installed. Please install it with `pip install llama-index[azure]`"
+        )
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)